control/metrics.go

```diff
@@ -250,6 +262,62 @@ func (m *metricType) Unit() string {
 	return m.unit
 }
 
+type catalogedPlugin struct {
```

Comment:
I was specifically asking if there was another one in the control package that implemented it. It seems like there would be.